pkg/endpoint/bpf.go

```diff
@@ -407,25 +408,64 @@ func updateCT(owner Owner, e *Endpoint, epIPs []net.IP,
 	return wg
 }
 
-// allocateRedirects must be called while holding the endpoint and consumable
+// addNewRedirectsFromMap must be called while holding the endpoint and consumable
 // locks for writing. On success, returns nil; otherwise, returns an error
 // indicating the problem that occurred while adding an l7 redirect for the
 // specified policy.
-func (e *Endpoint) allocateRedirects(owner Owner, m policy.L4PolicyMap) error {
+func (e *Endpoint) addNewRedirectsFromMap(owner Owner, m policy.L4PolicyMap, requiredRedirects map[string]struct{}) error {
 	for k, l4 := range m {
 		redirect := uint16(l4.L7RedirectPort)
 		if l4.IsRedirect() && redirect == 0 {
-			redirect, err := e.addRedirect(owner, &l4)
+			redirect, err := owner.UpdateProxyRedirect(e, &l4)
 			if err != nil {
 				return err
 			}
 			l4.L7RedirectPort = int(redirect)
 			m[k] = l4
+
+			proxyID := e.ProxyID(&l4)
+			if e.realizedRedirects == nil {
+				e.realizedRedirects = make(map[string]struct{})
+			}
+			e.realizedRedirects[proxyID] = struct{}{}
+			requiredRedirects[proxyID] = struct{}{}
```

Comment:
Not a big thing but if we make these map[string]bool then membership checks would simply be
`if found := requiredRedirects[anID]; found {` (I think we're using both maps as sets but I haven't checked `realizedRedirects`)

Reply:
+1

Reply:
I personally prefer `struct{}`, as the semantics is clearer, but I don't feel strongly about this.
Changed to `bool`.